src/ode_solver/JFNK_solver/jacobian_vector_product.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __JACOBIAN_VECTOR_PRODUCT__
+#define __JACOBIAN_VECTOR_PRODUCT__
+
+#include "dg/dg.h"
+
+namespace PHiLiP {
+namespace ODE{
+
+//template things
+//UPDATE WITH #IF STUFF
```

Comment:
add the stuff

Reply:
Sorry, I forgot to remove these comments. I'll replace with proper documentation for the class's functionality.